sdk/python/kfp/components/_components.py

```diff
@@ -381,16 +381,24 @@ def component_default_to_func_default(component_default: str, is_optional: bool)
         ('input_paths', Mapping[str, str]),
         ('output_paths', Mapping[str, str]),
         ('inputs_consumed_by_value', Mapping[str, str]),
+        ('input_uris', Mapping[str, str]),
+        ('output_uris', Mapping[str, str]),
     ],
 )
 
 
+def _not_implemented(name: str) -> str:
+    raise NotImplementedError
+
+
 def _resolve_command_line_and_paths(
     component_spec: ComponentSpec,
     arguments: Mapping[str, str],
     input_path_generator=_generate_input_file_name,
     output_path_generator=_generate_output_file_name,
     argument_serializer=serialize_value,
+    input_uri_generator=_not_implemented,
```

Comment:
nit: maybe type hint by the following? Similar to other parameters
```suggestion
    input_uri_generator: Callable[[str], str] = _not_implemented,
```

Reply:
Done.